Vassia srcchange

the-vdag opened the pull request:
Added the option of reading a txt point file for defining the seeding sources.

the-vdag committed:
Added Filepointstrategy for particle seeding

the-vdag committed:
Updated population schema to include the requirement for a file_points object

Updated population json schema to include the option for a file_points object and the required properties of this object

sgpearson17 reviewed (changes requested):
Hi Vassia, thanks for making a pull request with these additions- very handy! Can you provide an example file (both the yaml config file and an example csv file with points) for the main example model? Then I can test it properly. Let me know if you have questions.

the-vdag commented:
Hey Stuart,

Sure! I have attached the two files. The input data file is missing - let
me know if you want this also (it's a bigger file so I will send it via
filesender or similar if needed).

Thanks

Στις Δευ 17 Νοε 2025 στις 4:33 μ.μ., ο/η Stuart Pearson <
***@***.***> έγραψε:

> ***@***.**** requested changes on this pull request.
>
> Hi Vassia, thanks for making a pull request with these additions- very
> handy! Can you provide an example file (both the yaml config file and an
> example csv file with points) for the main example model? Then I can test
> it properly. Let me know if you have questions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sedtrails/sedtrails/pull/356#pullrequestreview-3473321303>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBGAU7L2ZB3EABHAWI2MUT35HTGDAVCNFSM6AAAAACJII3BDKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTINZTGMZDCMZQGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Vasiliki Dagalaki

T: +30 6949450415 / +31 0649122547

8000 6800
8000 6900
8100 6800
8100 6900
8200 6800
8200 6900
8300 6800
8300 6900
8400 6800
8400 6900
8500 6800
9500 7200
9500 7300
9600 7300
9700 7300
9700 7400
9800 7300
9800 7400
9900 7300
9900 7400
10000 7300
10000 7400
10100 7200
10100 7300
10100 7400
10200 7100
10200 7200
10200 7300
10200 7400
10300 6900
10300 7000
10300 7100
10300 7200
10300 7300
10400 6900
10400 7000
10400 7100
10400 7200
10500 6900
10500 7000
10900 6600
10900 6700
10900 6800
10900 6900
11000 6500
11000 6600
11000 6700
11000 6800
11000 6900
11100 6500
11100 6600
11100 6700
11100 6800
12100 7600
12200 7600
12200 8000
12300 7400
12300 7500
12300 7600
12300 8000
12400 7400
12400 7500
12400 7600
12400 7800
12400 7900
12400 8000
12500 6900
12500 7000
12500 7100
12500 7300
12500 7400
12500 7500
12500 7600
12500 7700
12500 7800
12500 7900
12600 6900
12600 7000
12600 7300
12600 7400
13000 7700
13000 7800
13000 7900
13100 7600
13100 7700
13100 7800
13100 7900
13100 8000
13200 7600
13200 7700
13200 7800
13200 7900
13200 8000
13300 7700
13300 7800
13300 7900
13300 8000
13400 7700
13400 7800
13400 7900
13400 8000
13500 7700
13500 7800
13500 7900
13500 8000
13600 7700
13600 7800
13600 7900
13600 8000
13700 7700
13700 7800
13700 7900
13800 7700
13800 7800
13800 7900
13900 7800
13900 7900
14000 7800
14000 7900
14100 7800
14200 7700
14200 7800
14300 7600
14300 7700
14300 7800
14400 7100
14400 7200
14400 7600
14400 7700
14400 7800
14500 7000
14500 7100
14800 6700
14800 6800
14900 6700
14900 6800
14900 6900
14900 7200
14900 7400
15000 7200
15000 7300
15000 7400
15100 7100
15100 7200
15100 7300
15100 7400
15200 6500
15200 6600
15200 7100
15200 7200
15200 7300
15200 7400
15300 6400
15300 6500
15300 6600
15300 7000
15300 7200
15300 7300
15300 7600
15300 7700
15400 6100
15400 6200
15400 6300
15400 6400
15400 6500
15400 7000
15400 7100
15400 7200
15400 7300
15400 7600
15400 7700
15500 6000
15500 6100
15500 6200
15500 6300
15500 6700
15500 6800
15500 6900
15500 7000
15500 7100
15500 7200
15500 7600
15500 7700
15600 5900
15600 6000
15600 6100
15600 6500
15600 6600
15600 6700
15600 6800
15600 7000
15600 7100
15600 7600
15600 7700
15700 6400
15700 6500
15700 6600
15700 6700
15700 6800
15700 6900
15700 7000
15700 7500
15700 7600
15700 7700
15800 6500
15800 7500
15800 7600
15800 7700
15900 7400
15900 7500
15900 7600
15900 7700
16100 7200
16200 6400
16200 6500
16200 6600
16200 6700
16200 7000
16200 7100
16200 7200
16200 7300
16300 6400
16300 6500
16300 7000
16300 7100
16300 7200
16400 6900
16400 7000
16400 7100
16400 7200
16700 5900
16700 6200
16800 5700
16800 5800
16800 5900
16800 6200
16800 6300
16900 5700
16900 5800
16900 5900
16900 6000
16900 6100
16900 6200
16900 6300
16900 6400
17000 4900
17000 5000
17000 5800
17000 5900
17000 6000
17000 6100
17100 4900
17100 5000
17100 5300
17100 5400
17100 5500
17100 5600
17200 5300
17200 5400
17200 5500
17200 5600
17200 5700
17300 4700
17300 5300
17300 5400
17300 5500
17300 5600
17300 5700
17400 4700
17400 4800
17400 5300
17400 5400
17400 5500
17400 5600
17400 5700
17500 4700
17500 4800
18000 4800
18000 4900
18000 5000
18100 4800
18100 4900
18100 5000
18200 4900
18400 4900
18400 5000
18500 4800
18500 4900
18500 5000
18600 4800
18600 4900
18900 4500
18900 4600
18900 4700
18900 4800
19000 4400
19000 4500
19000 4600
19000 4700
19100 4600
19100 4700

sgpearson17 commented:
Hi Vassia, I don't think the file attachment worked- can you either commit it here or email it to me directly?

Thanks!

Stuart

sgpearson17 committed:
Update example config and add source locations file

Renamed and revised the example YAML configuration for multisource particle release, updating input, output, and visualization settings. Added a new sources_xy_inlet.txt file specifying particle source coordinates for file-based release strategies.

sgpearson17 committed:
Add file_points seeding strategy documentation

Documents the new 'file_points' seeding strategy, including configuration parameters, usage examples, supported file formats, and data processing options. Updates both simulation parameters and user seeding documentation to describe how to release particles from externally specified coordinates.

sgpearson17 committed:
Add example text file for multisource seeding

Added a sample comma-separated text file with five points for use in the multisource seeding example. This provides users with a concrete example for configuring input sources in 'sedtrails-example-multisource.yaml'.

sgpearson17 committed:
Add tests for FilePointsStrategy in particle seeder

Introduces comprehensive unit tests for the FilePointsStrategy, covering basic usage, header/no-header handling, bounding box filtering, stride, deduplication, error cases, and integration with ParticleFactory. These tests improve coverage and robustness for file-based particle seeding functionality.

sgpearson17 reviewed (approved):
Thanks so much for this useful new feature, Vassia! I have gone through and tested your new seeding strategy and can confirm that it works as expected. I also added some documentation and tests. It would be great if you can also add similar tests and docs when making future pull requests (co-pilot helps!). I am still learning a lot about this too.

sgpearson17 committed:
Change simulation config file for execution

changed to handle merge conflict

sgpearson17 committed:
Fixed ruff errors

Removed unused numpy import and improved bbox dict parsing for clarity. Updated zip usage to enforce strict pairing of x and y coordinates, and simplified exception handling for CSV reading.

sgpearson17 committed:
Merge branch 'vassia_srcchange' of https://github.com/the-vdag/sedtrails into pr/356
